lib/cluster_holding.rb

```diff
@@ -24,4 +24,41 @@ def move(new_cluster)
     end
   end
 
+  # Updates a matching holding or adds it
+  def update
+    c = Cluster.find_by(ocns: @holding.ocn)
+    return cluster unless c
+
+    old_holding = c.holdings.to_a.find do |h|
+      h == @holding &&
+        h.date_received != @holding.date_received
```

Comment:
Do you think it's worth defining `#===` to mean "equal including the date"? Might be nice if we extend which attributes count/don't count.